### PR TITLE
applet.interface.jtag_svf: add support for manual state transitions.

### PR DESCRIPTION
**Pull request by wanda-phi**
Also does a bit of refactoring in `jtag-probe` and adds some hooks.

### CODE REVIEW

File: software/glasgow/applet/interface/jtag_probe/__init__.py [hunk 1]
```diff
@@ -547,6 +570,24 @@ async def enter_update_dr(self):
             self._state_error(JTAGState.DRUPDATE)
         self._state = JTAGState.DRUPDATE
 
+    async def state_path(self, path):
```
Comment (whitequark) [inline software/glasgow/applet/interface/jtag_probe/__init__.py:573]: `traverse_state_path`? All of the rest start with a verb, eg `enter_pause_dr`.

Comment (wanda-phi) [reply software/glasgow/applet/interface/jtag_probe/__init__.py:573]: Sounds good, thanks; I couldn't find a good name.

**Review by whitequark — COMMENTED** [review]
Looks good other than the comment.

**Review by whitequark — APPROVED** [review]
LGTM, thank you!

**Pull request merged**